<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>scNetViz - Single-cell RNASeq analysis</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
                
<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section id='title'>
    <h1>scNetViz - Single-cell RNASeq analysis</h1>
    <p><a href="http://www.rbvi.ucsf.edu/cytoscape/scNetViz/index.shtml">scNetViz</a> is a Cytoscape app designed to support the biological interpretation of scRNAseq experiments. It is available from the <a href="http://apps.cytoscape.org/apps/scnetviz" target="_blank">Cytoscape App Store</a>.</p>
     <br />
    <p>This tutorial includes two basic use cases, working with data from the EMBL-EBI Single Cell Expression Atlas, or local data. After loading the data into Cytoscape from local files or through the browser, we will generate plots from the data, perform differential expression analysis, generating networks from the top genes, and functionally characterize and visualize the networks.</p>
    <br />
    
<div class="footer" style="position: absolute;left: -1em;bottom: -3em; ">
    <img class="plain" src="scnetviz-logo.png" style="height:80px;opacity:0.5"> 
    <version>1.0.5</version>
</div>
</section>
                
<section>
        <h1>Setup</h1>
        <ul>
            <li>Install and launch the latest version of <a href="http://www.cytoscape.org/">Cytoscape</a>.</li>
            <li>Install the <b>scNetViz app</b> and <b>stringApp</b> via <highlight>Apps &rarr; App Manager</highlight>.</li>
        </ul>
             
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>

<section>
<section>
         <h1>Use Case 1: Load Data via scNetViz Browser</h1>
         <ul>
             <li>Launch Cytoscape. In the <highlight>Apps</highlight> menu, select <highlight>scNetViz &rarr; Load Experiment &rarr; From Single Cell Expression Atlas...</highlight>, or click the <img class="plain" src="sc_atlas_logo.png"> icon in the Cytoscape Toolbar.</li>
             <li>In the <highlight>Single Cell Experiment Atlas</highlight> (SCEA), click on the column header for the <highlight>Accession</highlight> column to sort it.</li>
             <li>Locate the experiment with acession number <b>E-GEOD-81383</b> and click on the row to highlight it. Click <highlight>View Data</highlight> to load the data.</li>  
         </ul>
         <p><img class="plain" src="SCEA-browser.png" style="width:60%"></p>
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
             
<section>
    <h1>Calculate Differential Expression</h1>
        <p>The data opens the <highlight>Experiment Data Table</highlight>, which contains normalized scRNA-seq quantification in transcripts per million in the <highlight>TPM</highlight> tab, metadata like experimental variables etc in the <highlight>Categories</highlight> tab, and for SCEA it also includes the grouping of cells as calculated by unsupervised clustering. The results displayed are for different values of the <b>K</b> parameter.</p>
        <ul>
            <li>In the <highlight>Categories Tab</highlight> locate the clustering result with a <b>sel.K</b> value of <b>True</b>. This represents the best clustering result for the data.</li>
            <li>Click the <highlight>Calculate Diff Exp</highlight> button to calculate differential expression.</li>
        </ul>         
         <p><img class="plain" src="SCEA-categories.png" style="width:70%"></p>
             
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
             
<section>
    <h1>Create Networks</h1>
         <p>The <highlight>DiffExp</highlight> tab will now be populated with data for the chosen cluster result.</p>
         <img class="plain" src="DiffExp.png" style="width:60%">
         <ul>
            <li>Click the <highlight>Create Networks</highlight> button to create networks. This will create networks by querying the <a href="https://string-db.org/">STRING database</a> for the protein products of genes that satisfy the <highlight>Network Analysis</highlight> cutoffs based on <b>FDR</b> and <b>Log2FC</b>, or <b>Max genes</b> count.</li>
         </ul>
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section> 
                
<section>
    <h1>Create Networks</h1>
         <p>In this case, 6 networks are created; one for each cluster and one for all the clusters combined. The <b>Log 2 Fold Change</b> for each cluster is visualized as the node fill color. In the <highlight>Results Panel</highlight>, two tabs will be loaded, for <b>scNetViz</b> and <b>stringApp</b>.</p>
         <img class="plain" src="Networks.png" style="width:65%">
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
    
<section>
    <h1>Functional Enrichment Analysis</h1>
         <ul>
         <li>Click on one of the clusters in the <highlight>Network Panel</highlight> to select it.</li>
         <li>In the <highlight>Get Enrichment</highlight> section of the <highlight>ScNetViz</highlight> tab, leave the default selections as-is and click <highlight>Retrieve Table</highlight>.</li>
         <li>The functional enrichment results will be loaded in a new <highlight>STRING Enrichment</highlight> panel in the <highlight>Table Panel</highlight> below the network.</li>
         <li>Click the <highlight>Draw charts using default color palette</highlight> button <img class="plain" src="ShowCharts.png">.</li>
         <li>Repeat these steps for the remaining networks.</li>
         </ul>
         <img class="plain" src="cluster4.png" style="width:33%">

         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>

<section>
    <h1>Compare and Contrast</h1>
    <ul>
      <li>Explore the 5 networks.  Note that this data set includes three melanoma cell lines.
        <ul>
          <li>Do the enriched terms make sense?</li>
          <li>Do you see any differences between the networks?</li>
          <li>Do you see any similarities between the networks?</li>
          <li>Look at the enriched terms for clusters 2,3, and 4 and compare with clusters 1 and 5.  See any consistencies or differences?</li>
        </ul>
      </li>
    </ul>
</section>
</section>


<section>               
<section>
    <h1>Use case 2: Load a Local scRNA-Seq Dataset</h1>
                
    <p>In this workflow, we will load normalized data and cluster assignments from local files. 
    <ul>
        <li>Download the <a href="https://github.com/cytoscape/cytoscape-tutorials/raw/gh-pages/protocols/data/E-GEOD-109979-normalised-files.zip">normalized data</a> (zipped folder).</li>
        <li>Download the <a href="https://raw.githubusercontent.com/cytoscape/cytoscape-tutorials/gh-pages/protocols/data/E-GEOD-109979.clusters.tsv">cluster assignments</a>. When it opens in the browser, right-click and select <highlight>File &rarr; Save Page As...</highlight> (Chrome, Firefox), <highlight>File &rarr; Save As...</highlight> (Safari).</li>
    </ul>
        
    
<aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
               
<section>
    <h1>Load data</h1>
    <ul>
    <li>Launch Cytoscape. In the <highlight>Apps</highlight> menu, select <highlight>scNetViz &rarr; Load Experiment &rarr; Import from file...</highlight>.</li>
    <li>Select <b>Homo sapiens</b> as the species, by typing directly into the drop-down. Browse to find the <b>E-GEOD-109979</b> zip file under <highlight>File or directory with MTX matrix and headers</highlight>. Click <b>OK</b> to load.</li>
    <li>Leave <highlight>Skip first line of header files</highlight> blank.</li>
    </ul>
           
    <!--
    <img class="plain" src="Categories.png" style="width:60%"> -->
<aside class="notes" data-markdown>
         Notes go here 
         </aside>
    
</section>
                
<section>
    <h1>Load Categories</h1>
    <ul>
    <li>In the <highlight>Experiment Data Table</highlight>, click the <highlight>Add Category</highlight> drop-down and select <highlight>Import from file...</highlight>.</li>
    <li>Browse to select the .csv file with category data, and leave remaining selections as the default.</li>
    </ul>
    <img class="plain" src="Categories.png" style="width:70%">
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
    
<section>
    <h1>Load Categories</h1>
    <p>Once the categories are loaded, the data viewer is updated:</p>
    
    <img class="plain" src="Categories-viewer.png" style="width:70%">
    <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
                
<section>
    <h1>Create UMAP Plot</h1>
    <ul>
        <li>Under <highlight>New Cell Plot</highlight>, select <b>UMAP</b>. Keep the default parameters.</li>
    
    </ul>
   <p>
    <img class="plain" src="Usecase2-UMAP.png" style="width:70%">
    </p>
         <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>

<section>
    <h1>Calculate Differential Expression</h1>
    
    <ul>
    <li>In the <highlight>Categories Tab</highlight> locate the clustering result with a <b>sel.K</b> value of <b>True</b>. This represents the best clustering result for the data.</li>
    <li>Click the <highlight>Calculate Diff Exp</highlight> button to calculate differential expression.</li>
    </ul>
    
    <p>
    <img class="plain" src="Usecase2-diffexp.png" style="width:70%">
    </p>
    <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
                
<section>
    <h1>Create heatmap</h1>
    <p>Next, we will generate a heatmap showing the top differentially expressed genes:</p>
    <ul>
        <li>Under <highlight>View Plots</highlight>, select <b>Heatmap</b>.</li>
    </ul>
    <p>
    <img class="plain" src="Usecase2-heatmap.png" style="width:70%">
    
    </p>
    <aside class="notes" data-markdown>
         Notes go here 
         </aside>
</section>
                
<section>
    <h1>Create Networks</h1>
    <ul>
        <li>In the <highlight>DiffExp</highlight> tab, click the <highlight>Create Networks</highlight> button to create networks from the <a href="https://string-db.org/">STRING database.</a></li>
    </ul>
    
    <p>In this case, 5 networks are created; one for each cluster and one for all the clusters combined. The <b>Log 2 Fold Change</b> for each cluster is visualized as the node fill color.</p>
    
    <img class=plain src="cluster2-2.png" style="width:42%">


    
  <aside class="notes" data-markdown>
         Notes go here 
         </aside>
                
</section>
</section>            
          
<section>
    <h1>Exporting Networks</h1>
    <p>Cytoscape provides a number of ways to export results and visualizations:</p>
	<ul>
        <li>As an image: <highlight>File &rarr; Export &rarr; Network to Image...</highlight></li>
        <li>To a public repository: <highlight>File &rarr; Export &rarr; Network to NDEx</highlight>, or <highlight>File &rarr; Export &rarr; Collection to NDEx</highlight></li>
		<li>As a Cytoscape JSON file: <highlight>File &rarr; Export &rarr; Network to File</highlight> and select <b>Cytoscape.js JSON</b> as the format.</li>
	</ul>
</section>

                
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>

<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>  

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },
				
				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
