<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>WikiPathways App</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">

<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section id='title'>
    <h1>WikiPathways App</h1>
    <p><a href="https://www.wikipathways.org" target="_blank">WikiPathways</a> is a database of biological pathways maintained by and for the scientific community. Each pathway model in the database is annotated with database identfiers, which can be used for computation and data visualization. The WikiPathways App for Cytoscape is available from the <a href="http://apps.cytoscape.org/apps/wikipathways" target="_blank">App Store</a>.</p>
    <p>This tutorial consists of two workflows:</p>
    <ul>
        <li>Basic workflow describing pathway import and data visualization, including importing the pathway as a network, and creating a legend.</li>
        <li>Advanced workflow describing how to visualize multiple data points on a set of pathways of interest.</li>
    </ul>
    <br />
<div class="footer" style="bottom: -3em;">
    <a href="https://github.com/cytoscape/cytoscape-tutorials/wiki"><img class="plain" src="../../assets/images/cytoscape-tutorials.png" style="height:70px;opacity:0.5"></a></div>

</section>

<section>
        <h1>Setup</h1>
        <ul>
            <li>Install and launch the latest version of <a href="http://www.cytoscape.org/">Cytoscape</a>.</li>
            <li>Install the <a href="http://apps.cytoscape.org/apps/wikipathways" target="_blank">WikiPathways</a> and <a href="http://apps.cytoscape.org/apps/legendcreator" target="_blank">Legend Creator</a> apps via <highlight>Apps &rarr; App Store &rarr; Show App Store</highlight>.</li>
	</ul>

         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section>
<section>
         <h1>Basic Workflow: Importing Pathways</h1>
         <ul>
             <li>Launch Cytoscape, and in the <highlight>Network Search Tool</highlight> at the top of the <highlight>Network</highlight> tab in the <highlight>Control Panel</highlight>, select <highlight>WikiPathways</highlight> in the drop-down. Type in <b>"cell cycle"</b> (with quotes) and click the search icon.</li>
             <li>A list of results for all available species will open. To see results from a select species, click the <highlight>Only</highlight> checkbox and select a species from the drop-down (example: Homo sapiens).</li>
             <li>Select the human <b>Cell Cycle</b> pathway (WP179) and click <highlight>Import as Pathway</highlight>, or simply double-click the pathway in the list.</li>
         </ul>
         <img src="WPApp-results.png" style="width:55%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section>
    <h1>Importing Pathways</h1>
         <p>The pathway opens in Cytoscape with the original pathway layout and the WikiPathways style.</p>
	 <p>Tip: if the labels are not depicted in the visualization,  click on the diamond shaped button below the visualization to enable showing names. This step is depicted in the figure below in the red square.</p>
         <img src="CellCycle.png" style="width:70%">

         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section>
    <h1>Pathway Node Identifiers</h1>
         <p>Nodes in pathways from WikiPathways are annotated with identifiers from public databases, visible in the <highlight>XrefId</highlight> column in the <highlight>Node Table</highlight>. During import, the WikiPathways App maps these identifiers automatically to Ensembl identifiers for genes and proteins and ChEBI for metabolites, adding them to a new column (to the far right of the <highlight>Node Table</highlight>).</p>
         <img src="CellCycleMapped.png" style="width:50%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
         </section>

<section>
    <h1>Data Import</h1>
         <p>The experimental data that we will use is a lung cancer dataset from <a href="https://www.cancer.gov/types/lung" target="_blank">TCGA</a>, comparing lung cancer biopses versus normal tissue. This dataset is annotated with Ensembl identifiers, so we can use the mapped Ensembl column to connect to the dataset:</p>
    
         <ul>
         <li>Load the <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis-visualization/ExampleData/lung.expr.csv">lung.expr.csv</a> file under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>; select the file you want to import in the menu, and click on 'Open'. Alternatively, drag and drop the data file directly onto the <highlight>Node Table</highlight>.</li>
         <li>Under <highlight>Where to Import Table Data</highlight>, select <highlight>To a Network Collection</highlight> and select the relevant network collection in the second drop-down; if you only have one network opened in Cytoscape, this network is selected automatically.</li>
         <li>In the <highlight>Key Column for Network</highlight> drop-down, select the new <b>Ensembl</b> column.</li>
         <li>In the <highlight>Preview</highlight> table, note that the <b>Gene ID</b> column is already selected as the key.</li>
	 <li>The next slide in this presentation shows an example of the data import interface.</li>
         </ul>

         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section>
    <h1>Data Import</h1>
         <p>The data import interface should now look like this:</p><br />
         <ul>
             <div style="width:40%;float:left; margin-right: 40px">
					<img src="WPDataImport.png">
				</div>
         <li>Click <highlight>OK</highlight> to import.</li>
         <li>Once import is complete, look for the newly imported data columns in the <highlight>Node Table</highlight> to confirm that it worked.</li>
	     <div style="width:30%;float:right; margin-right: 40px">
					<img src="WPDataImportSuccess.png">
				</div>		 
         </ul>
         <div style="position:absolute;left:28em;bottom:1em;">
             <p><b><a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/importing-data-from-tables/#/" target="_blank">Learn more about Importing Data</a></b></p>
         </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section>
         <h1>Data Visualization: Node Fill Color</h1>
         <p>We now have the data in Cytoscape, linked to the pathway, and ready to map to visual properties of the nodes in the network. We are going to visualize the fold change as <b>Node Fill Color</b>.</p>
         <ul>
         <li>In the lefthand side panel, select the <highlight>Style</highlight> tab of the <highlight>Control Panel</highlight>. For the <highlight>Fill Color</highlight> visualization option, select the middle column (<highlight>Map.</highlight> box).</li>
         <li>In the <highlight>Column</highlight> drop-down, select <b>"log2FC"</b>.</li>
         <li>In the <highlight>Mapping Type</highlight> drop-down, select <highlight>Continuous Mapping</highlight>. This will create a default color gradient from blue (negative fold change) to red (positive fold change):</li>
    </ul>
         <img class="plain" src="CellCycleData.png" style="width:60%">

</section>
    
<section>
    <h1>Data Visualization: Node Border Width</h1>
        <p>We also have a significance measure for the fold change, which we can add to our visualization. For this, we will use both <highlight>Node Border Width</highlight> and <highlight>Node Border Size</highlight> to visualize the p value.</p>
        <ul>
            <li>Click on the <highlight>Map.</highlight> cell for the <highlight>Node Border Width</highlight> property in the <highlight>Style</highlight> panel.</li>
            <li>In the <highlight>Column</highlight> drop-down, select <b>P.value</b>.</li>
            <li>In the <highlight>Mapping Type</highlight> drop-down, select <highlight>Continuous Mapping</highlight>.</li>
            <li>Double-click on the gradient, which defines the node border width over the range of p values.</li>
            <li>First, we will define the min/max of the range of p values we are interested in, that is anything under 0.05. Click on <highlight>Set Min and Max...</highlight> and set the max to <b>0.05</b>. Click <highlight>OK</highlight> to save.</li>
            <li>Click on the handle for the max value (black triangle on the right-most end of the gradient) and set the value to <b>5</b> in the <highlight>Node Border Width</highlight> field. Repeat with the handle for the min value. Click <highlight>OK</highlight> to update the mapping.</li>
            <li>Click the <highlight>Def.</highlight> value for <highlight>Node Border Paint</highlight> and select a dark grey color.</li>
        </ul>
        <img src="NodeWidthGradientWP.png" style="width:40%">
    </section>
    
    
<section>
         <h1>Data Visualization: Node Border Color</h1>
         <ul>
            <li>Click on the <highlight>Map.</highlight> cell for <highlight>Node Border Paint</highlight> in the <highlight>Style</highlight> panel.</li>
            <li>In the <highlight>Column</highlight> drop-down, select <b>P.value</b>.</li>
            <li>In the <highlight>Mapping Type</highlight> drop-down, select <highlight>Continuous Mapping</highlight>.</li>
            <li>Double-click on the gradient.</li>
            <li>Similar to the last step, we will define the min/max of the range of p values we are interested in, that is anything under 0.05. Click on <highlight>Set Min and Max...</highlight> and set the max to <b>0.05</b>. Click <highlight>OK</highlight> to save.</li>
            <li>Click on the large triangle handle at the right-most end of the gradient. This defines the border color for nodes with <b>P.value</b> > 0.05, meaning not significant, so let's set it to dark grey.</li>
            <li>Next, set all the other handles to the same bright pink color. We can delete the middle handle since we want the same color for anything under 0.05, simply select it and click <highlight>Delete</highlight>.</li>
         </ul>
         
    <img src="NodeBorderGradientWP.png" style="width:40%">
         

         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
    
<section>
    <h1>Data Visualization</h1>
    <p>We now have a data visualization of our data on the pathway, with fold change represented as a gradient node fill color and significance as node border color/width.</p>
    
    <img class="plain" src="CellCycleDataFinal.png" style="width:70%">
    
</section>
    
<section>
    <h1>Importing Pathways as Networks</h1>
    <p>In addition to importing pathways with the visual style and layout unique to WikiPathways, the WikiPathways App also allows import of any pathway as a simple network.</p>
        <ul>
	     <li>In the left-hand side menu, move away from the <highlight>Style</highlight> Panel back to the <highlight>Network</highlight> Panel.</li>
             <li>Again, select <highlight>WikiPathways</highlight> in the drop-down of the <highlight>Network Search Tool</highlight> and search for <b>"cell cycle"</b> (with the quotes) and click the search icon.</li>
             <li>In the list of results, select the <b>human</b> pathway and click <highlight>Import as Network</highlight>.</li>
	     <li>We can now repeat the data import and style selection to create a visualization like in the previous pathway:</li>
         </ul>
         <img class="plain" src="CellCycleNetworkData2.png" style="width:46%">

</section>
    
<section>
    <h1>Adding a Legend</h1>
    <p>To add a legend to your pathway visualizations, you can use the <b>Legend Creator</b> App.</p>
    <ul>
        <li>In the <highlight>Network Panel</highlight> select the network you want to add a legend to. For this example, let's use the Cell Cycle pathway with the original pathway layout.</li>
        <li>In the <highlight>Control Panel</highlight>, select <highlight>Legend Panel</highlight> either from the drop-down at the top, or by clicking on its tab.</li>
        <li>The <highlight>Legend Panel</highlight> will have some visual properties selected by default. Make sure they are the ones you want to include in the legend. For this example, <b>Node Fill Color</b>, <b>Node Border Width</b> and <b>Node Border Paint</b>.</li>
        <li>Select a <highlight>Title</highlight> for the legend, for example "Lung cancer vs normal". For this example, the <highlight>Subtitle</highlight> is left blank.</li>
        <li>Click <highlight>Refresh Legend</highlight> to add the legend to your network.</li>
    
    </ul>
</section>
<section>
    <h1>Adding a Legend</h1>
    <p>Our final visualization now includes a legend. In this case, the three separate legend graphics were resized from the default. This can be done by first clicking <highlight>Toggle Annotation Selection</highlight> <img class="plain" src="ToggleAnnotationSelection.png"> at the bottom of the <highlight>Network View Window.</highlight></p>
    <img class="plain" src="CellCycleDataFinalLegend.png" style="width:60%">
</section>
    
<section>
    <h1>Saving and Exporting Data Visualization</h1>
    <p>Cytoscape provides a number of ways to save results and visualizations:</p>
	<ul>
        <li>As a session: <highlight>File &rarr; Save Session</highlight>, <highlight>File &rarr; Save Session As...</highlight></li>
        <li>As an image: <highlight>File &rarr; Export &rarr; Network to Image...</highlight>, selecting for example <b>pdf</b> or <b>png</b> from the <highlight>Export File Format</highlight> drop-down.</li>
        <li>To a public repository: <highlight>File &rarr; Export &rarr; Network to NDEx</highlight></li>
	</ul>
</section>
    
</section>
                
<section>
    <section>
        <h1>Advanced Workflow: Importing Multiple Pathways</h1>
        <p>This workflow describes how to work with multiple WikiPathways pathways in Cytoscape, and how to visualize more than one data point. This analysis assumes that we already have a list of pathways of interest. This list could come from enrichment analysis, but could also be a list of pathways of interest from prior knowledge. </p>
        <p>The data we will be using for this workflow (<a href="https://raw.githubusercontent.com/cytoscape/cytoscape-tutorials/gh-pages/protocols/data/E-GEOD-68086.csv">E-GEOD-68086.csv</a>) is from a study investigating the RNA-seq signatures of platelets from patients with various types of cancer, in an effort to develop a platelet-based diagnostic (<a href="https://www.cell.com/cancer-cell/fulltext/S1535-6108(15)00349-9" target="_blank">Best et el., 2015).</a> The data was downloaded from <a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-68086/Downloads" target=_blank>Expression Atlas</a>.</p>
        <ul> 
        <li>If you are continuing from the previous workflow, make sure the session is saved and then select <highlight>File &rarr; Close Session</highlight> to start from scratch.</li>
        <li>Using the the <highlight>Network Search Tool</highlight> with <highlight>WikiPathways</highlight> selected, search for and open the following <b>human</b> pathways, with the "Import as Pathway" option. <b>Pro-tip</b>: You can search for the WPID instead of the title.
            <ul>
                <li>WP179: Cell cycle</li>
                <li>WP2446: Retinoblastoma gene in cancer</li>
                <li>WP4830: GDNF/RET signaling axis</li>
                <li>WP707: DNA damage response</li>
            </ul></li>
        </ul>
    </section>
    <section>
    <p>The 4 pathways will be listed as separate network collections in the <highlight>Network Panel</highlight>:</p>
    <img src="WP-MultiplePathways.png">
</section>
    <section>
        <h1>Data Import</h1>
        <p>Now that we have our pathways of interest open, we can load the data. Cytoscape allows you to import data either to a specific network, collection of networks, or a set of networks. We will use the latter, to avoid having to load the data multiple times.</p>
        <ul>
            <li>Load the <a href="https://raw.githubusercontent.com/cytoscape/cytoscape-tutorials/gh-pages/protocols/data/E-GEOD-68086.csv">E-GEOD-68086.csv</a> file under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>.</li>
         <li>Under <highlight>Where to Import Table Data</highlight>, select <highlight>To selected networks only</highlight> and click the <b>Select All</b> button to select all of the networks in the <b>Network List</b>.</li>
	 <li>For the <highlight>Import Data as:</highlight> option, select the <b>Ensembl</b> column.</li>
         <li>In the <highlight>Key Column for Network</highlight> drop-down, select the <b>Node Table columns</b>.</li>
         <li>In the <highlight>Preview</highlight> table, note that the <b>Gene ID</b> column is already selected as the key.</li>
        </ul>
        
        <div class="fragment">
        <p>Each network should now have multiple extra columns corresponding to fold change and p-value for several cancer types:
        </p>
        <img src="WP-DataNodeTable.png" style="width:60%">
        </div>
    </section>
    
    <section>
         <h1>Data Visualization</h1>
         <p>To visualize the data on the pathways, we are going to use Cytoscape's built-in function for <b>Custom Graphics</b> to visualize the fold change values for three cancer types: breast carcinoma, pancreatic adenocarcinoma and colorectal carcinoma. For this particular worflow, the visualized data points should be of the same type, for example fold changes.</p>
         <ul>
         <li>In the Style menu (left-hand side panel), click on the <highlight>Def.</highlight> column of <highlight>Image/Chart 1</highlight>.</li>
        <li>Under <highlight>Charts</highlight> select <highlight>Heatmap</highlight> option (third icon from the left). Select the columns you want to visualize; <b>breast carcinoma vs normal foldChange</b>, <b>pancreatic adenocarcinoma vs normal foldChange</b> and <b>colorectal carcinoma vs normal foldChange</b>. Remove any other columns selected by default.</li>
        <li>Under <highlight>Customize</highlight>, select <highlight>Horizontal Orientation</highlight>.</li>
        <li>Under <highlight>Color Palette</highlight> select <b>HSV Red-Blue</b> (left-most).</li>
        <li>Click the <highlight>Not Available</highlight> icon <img class="plain" src=NotAvailable.png> in the color palette and change its color from dark grey to white.</li>
        <li>Click <highlight>Apply</highlight>.</li>

         </ul>
</section>
<section>
    <h1>Data Visualization</h1>
    <p>The three data points are now visualized as vertical stripes, each with a gradient from blue-red for fold change. Note that for this particular graphic (heatmap with horizontal orientation), the data points are displayed in the opposite order from how they were listed in the Custom Graphics interface, so in this case from <b>left to right</b> it is showing <b>pancreatic adenocarcinoma vs normal</b>, <b>colorectal carcinoma vs normal</b> and <b>breast carcinoma vs normal</b>.</p>
    <img src="HeatmapCancerTypes.png" style="width:50%">
</section>
    
<section>
    <h1>Data Visualization</h1>
    <p>The same visualization will be automatically applied to the remaining pathways once you click on them:</p>
    <div class="r-stack">
  <img src="Multi4.png" width="450" >
  <img src="Multi1.png" width="450" >
  <img src="Multi2.png" width="450" >
  <img src="Multi3.png" width="450" >
  
</div>
    
</section>

<section>
    <h1>Saving and Exporting Data Visualization</h1>
    <p>Cytoscape provides a number of ways to save results and visualizations:</p>
	<ul>
        <li>As a session: <highlight>File &rarr; Save Session</highlight>, <highlight>File &rarr; Save Session As...</highlight></li>
        <li>As an image: <highlight>File &rarr; Export &rarr; Network to Image...</highlight>, selecting for example <b>pdf</b> or <b>png</b> from the <highlight>Export File Format</highlight> drop-down.</li>
        <li>To a public repository: <highlight>File &rarr; Export &rarr; Network to NDEx</highlight></li>
	</ul>
</section>
    
</section>
                
<section>
    <h1>Downstream Analysis Options</h1>
    <p>Whether you import pathways as networks or as pathways, there are many different downstream analysis options available in Cytoscape:</p>

    <ul>
        <li>Extending pathways with regulatory information using <a href="https://cytargetlinker.github.io/pages/tutorials/tutorial2" target="_blank">CyTargetLinker</a></li>

    </ul>


</section>


<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>

<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },

				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
