<section>
	<h1>Proteomics</h1>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <p style="text-align:left">Caveats:
  <ul>
    <li>I am not an expert</li>
    <li>This is just some of the basics, there are multiple techniques, different kinds of detecters, different kinds of mass spec machines, etc.</li>
    <li>The goal is just to provide some idea of how this works to help understand the data</li>
  </ul>
  </p>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <div>
    <img src="modules/proteomics/MassSpec1.png" width="80%">
  </div>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <div>
    <img src="modules/proteomics/CO2-spectra.png" width="80%">
  </div>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <h2 style="font-size:80%">Tandem Mass Spectrometry (MS/MS)</h2>
  <p style="text-align: left; font-size:80%">When dealing with proteins or lots of peptides, the m/z ratios might be too close to distinguish.  Tandem Mass Spectrometry addresses that by
  adding an additional fragmentation step</p>
  <div>
    <img class="fragment" src="modules/proteomics/MS_MS.png" width="80%">
  </div>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <h2 style="font-size:80%">Proteomics</h2>
  <div>
    <img src="modules/proteomics/3typesofproteomics.png" width="80%">
  </div>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <h2 style="font-size:80%">Proteomics</h2>
  <p style="text-align: left;font-size:80%">How do we get from spectra to peptide?</p>
  <div>
    <img src="modules/proteomics/peptide-identification.png" width="80%">
  </div>
  <p style="text-align: left;font-size:80%">Possible problem: peptide might not uniquely identify a protein.</p>
</section>

<section>
	<h1 style="font-size:100%">(Very) Brief Intro to Mass Spectrometry</h1>
  <p style="text-align:left">Resources for learning more:
  <ul>
    <li><a href="https://www.youtube.com/watch?v=hvKbAoKkSM8">Georgia Tech Video on Fundamentals of Mass Spectrometry</a></li>
    <li><a href="https://www.khanacademy.org/science/ap-chemistry-beta/x2eef969c74e0d802:atomic-structure-and-properties/x2eef969c74e0d802:mass-spectrometry-of-elements/v/mass-spectrometry">Kahn Academy Video on Mass Spectrometry</a></li>
    <li><a href="https://masspec.scripps.edu/learn/ms/">Scripps Center for Metabolimics</a>.  A more thorough discussion of current techniques.</li>
    <li><a href="https://chem.libretexts.org/Bookshelves/Organic_Chemistry/Supplemental_Modules_(Organic_Chemistry)/Spectroscopy/Mass_Spectrometry">LibreTexts Chemistry Chapter</a>.  Another pretty thorough discussion.</li>
    <li><a href="https://www.youtube.com/watch?v=wx4F6kGy1Fs">Simple overview of proteomics from Lars Juhl Jensen</a></li>
    <li><a href="https://www.youtube.com/watch?v=ZgwNWRul98o">More detailed presentation of proteomics from Lennart Martens</a></li>
    <li><a href="https://www.youtube.com/watch?v=2BDYcoVB45U&t=1s">EMBL-EBI Training presentation on proteomics</a></li>
  </ul>
</section>


<section>
  <h1 style="font-size:120%">Understanding Proteomics Results</h1>
  <p>Overview of analytical pipeline</p>
  <div>
    <img src="modules/proteomics/pipeline.jpg" >
  </div>
  <a style="font-size: large;" href="https://doi.org/10.3389/fmed.2021.747333">https://doi.org/10.3389/fmed.2021.747333</a>
</section>

<section>
  <h1 style="font-size:120%">Understanding Proteomics Results</h1>
  <p>Example dataset after identification and quantitation</p>
  <div>
    <img src="modules/proteomics/spreadsheet.png" title="add image here" >
  </div>
</section>

<section>
  <h1 style="font-size:120%">Analyzing Proteomics Data</h1>
  <p>R packages for proteomics analysis</p>
  <ul>
    <li><a href="https://bioconductor.org/packages/devel/BiocViews.html#___MassSpectrometry">MS packages at Bioconductor</a></li> 
    <li><a href="https://bioconductor.org/packages/devel/BiocViews.html#___Proteomics">Proteomics packages at Bioconductor</a></li> 
    <li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-023-05360-7">tidyproteomics</a> - an open-source R package and data object for quantitative proteomics post analysis and visualization</li> 
    <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10010602/">promor</a> - a comprehensive R package for label-free proteomics data analysis and predictive modeling</li> 
    <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10088014/">prolfqua</a> - a Comprehensive R-Package for Proteomics Differential Expression Analysis</li> 
  </ul>
</section>

<section>
  <h2>Analyzing PTM Data</h1>
    <p>Peptide identification must account for the masses of potential post-translational modifications (PTMs)</p>
    <div>
      <img src="modules/proteomics/ptm-spectra.png" >
    </div>
    <ul>
      <li><a href="https://bioconductor.org/packages/devel/bioc/html/MSstatsPTM.html">MSstatsPTM</a> - Statistical Characterization of Post-translational Modifications</li> 
    </ul>
  </section>

<section>
  <h2>Analyzing AP/MS or BioID Data</h1>
	<p>AP/MS and BioID results are a list of interactions</p>
    <div>
      <img src="modules/proteomics/ap-ms.png" >
    </div>
</section>
      

