<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Pathway Figure Gene Set Analysis</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">

<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section>
	<h1>Pathway Figure Gene Set Analysis</h1>
    <p style="text-align: left">Through a combination of optical character recognition, machine learning and manual curation, gene sets have been extracted from <a href="https://doi.org/10.1186/s13059-020-02181-2" target="_blank">published pathway figures</a>. 32,000 of these sets are available in <a href="https://ndexbio.org/#/networkset/85034b42-de8a-11ea-99da-0ac135e8bacf" target="_blank">NDEx</a>, and can be used directly in Cytoscape.<br /><br />
    This protocol describes how to visualize data on a gene set, create a network from it in two different ways, perform functional enrichment and include the original figure for comparison. <b>The starting point for this kind of analysis is a list of gene sets found to be enriched in an experiment, through functional enrichment analysis against the full set of gene sets.</b></p>
    
    <img class="plain" src="network-final.png" style="height:300px"> 
    
    <p style="text-align: left">If you don't already have the <b>stringApp</b> installed, you can install it from the <a href="http://apps.cytoscape.org/apps/stringapp" target="_blank">Cytoscape App Store</a> or from Cytoscape via <highlight>Apps &rarr; App Manager ...</highlight></p>
    <div class="footer" style="bottom: -2em;">
    <a href="https://github.com/cytoscape/cytoscape-tutorials/wiki"><img class="plain" src="../../assets/images/cytoscape-tutorials.png" style="height:70px;opacity:0.5"></a></div>
</section>
                
<section>
    <h1>Import Pathway Figure Gene Set</h1>
    
    <ul>
        <li>Navigate to <a href="https://ndexbio.org/#/networkset/85034b42-de8a-11ea-99da-0ac135e8bacf" target="_blank">Published Pathway Figure Analysis Set</a> at NDEx.</li>
        <li>In the <highlight>Network Name</highlight> field, type in <b>Possible interaction between PI3K/AKT/mTOR network and PIM in ovarian cancer</b> to find the network by name.</li>
        <li>Click on the network. The page will display the gene set on the left and the published figure they were extracted from on the right.</li>
        <li>Save the figure by right-clicking on it and selecting <highlight>Save Image As..</highlight>.</li>
        <li>To open the gene set into Cytoscape, click the <highlight>Open in Cytoscape</highlight> button in the lower right.</li>
    </ul>
    <p>The gene set will appear as a grid of nodes in Cytoscape, similar to what it looks like at NDEx.</p>
    <p>
    <img class="plain" src="pfocr-gene-set-plain.png" style="width:60%"></p>
</section>
                
<!--<section>
    <h1>Import NDEx Gene Set</h1>
    <p>We will start by importing a gene set from NDEx.</p>
    <ul>
        <li>Click the <highlight>Import Network from NDEx</highlight> <img class="plain" src="NDEx-search.png" style="height:20px"> icon in the control panel.</li>
        <li>In the NDEx <highlight>Find Networks</highlight> interface, type in the network UUID <b>bcd5afe8-de97-11ea-99da-0ac135e8bacf</b> and click the search icon.</li>
        <li>When the results load, double-click the corresponding entry to open the network, in this case the network with name <b>PMC5884799__41598_2018_24032_Fig7_HTML.jpg</b>.</li>
        <li>Close the NDEx interface.</li>
    </ul>
    <img class="plain" src="NDEx-results.png" style="width:45%">
</section>-->

<section>
<section>
    <h1>Data Integration and Visualization</h1>
    <p style="text-align: left">Next, we will import experimental data from <a href="https://cancergenome.nih.gov/" target="_blank">TCGA</a>, comparing gene expression between two molecular subtypes of ovarian cancer (Mesenchymal and Immunoreactive). The data is annotated with <b>NCBI Gene</b> identifiers, which is also what the nodes in the gene set are annotated with.</p>
    <ul>
        <li>Download a local copy of <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/data/TCGA-Ovarian-MesenvsImmuno_data.csv" target="_blank">TCGA-Ovarian-MesenvsImmuno_data.csv</a>.</li>
        <li>Load the <b>TCGA-Ovarian-MesenvsImmuno_data.csv</b> file under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>. Alternatively, drag and drop the data file directly onto the <highlight>Node Table</highlight>.</li>
	   <li>Select the <b>ncbigene</b> column as the <b>Key column for Network</b>, to match the <b>Gene</b> column of the data.</li>
        <li>To complete the import, click <highlight>OK</highlight>. Two new columns of data will be added to the <highlight>Node Table</highlight>.</li>
    </ul>
</section>   
                
<section>
    <h1>Data Integration and Visualization</h1>
    <p style="text-align: left">We can now add a simple visualization of the data on the set of nodes.</p>
    <ul>
        <li>In the <highlight>Style</highlight> interface, under <highlight>Node Fill Color</highlight>, create a continuous mapping for <highlight>logFC</highlight>, with the default ColorBrewer <b>Red-Blue</b> gradient.</li>
        <li>Set the default node fill color to light gray.</li>
        <li>The gene set is already arranged with a grid layout, but we can make it a bit more compact. Under <highlight>Layout &rarr; Settings...</highlight>, select the <highlight>Grid Layout</highlight>.</li>
        <li>Set the <highlight>Vertical spacing between nodes</highlight> to <b>50</b> and <highlight>Horizontal spacing between nodes</highlight> to <b>80</b>.</li>
    </ul>
    <img class="plain" src="pfocr-gene-set.png" style="width:50%">
</section>
                
<section>
    <h1>Add Pathway Figure as Annotation</h1>
    <p style="text-align: left">We can add the original published figure for this gene set as a Cytoscape annotation, to add context to the nodes.</p>
    <ul>
        <li>In the <highlight>Annotations</highlight> tab of the <highlight>Control Panel</highlight>, select the <highlight>Image</highlight> icon at the top and then click anywhere in the network view to place the annotation. The initial placement can easily be changed later.</li>
        <li>Select the pathway figure you downloaded previously in the file browser.</li>
        <li>Once the annotation is placed, enable annotation selection by clicking the <highlight>Toggle Annotation Selection</highlight> button <img class="plain" src="toggle-annotation.png" style="height:35px"> in the <highlight>Network View Tools</highlight> under the network.</li>
    </ul>
    <img class="plain" src="gene-set-final.png" style="width:85%">
</section>
    
<section>
    <h1>Manually Layout Gene Set Nodes</h1>
    <p style="text-align: left">Next, we will arrange the gene set nodes and add interaction with the goal of reproducing the information from pathway figure in a computational model. We can start by moving the nodes into the approximate location from the figure. Note that the label displayed on the node, <b>name</b>, is the official gene symbol, whereas the original published figures sometimes use various gene symbol aliases. Use the <b>symbol</b> column to match nodes to the figure.</p>
    <p style="text-align: left">In this example, the original figure contained several complexes and isoform groups that were illustrated as single nodes. In the gene set these have been expanded to include all members of isoforms and complex subunits.</p>
    <img class="plain" src="gene-set-layout.png" style="width:85%">

</section>
    
<section>
    <h1>Add Group Nodes to the Gene Set</h1>
    <p style="text-align: left">Cytoscape allows for grouping of nodes, which we can use to organize complexes and groups.</p>
    <ul>
    <li>Select the three IRS nodes by clicking and dragging.</li>
    <li>Right-click and in the context menu, select <highlight>Group &rarr; Group Selected Nodes</highlight>. Enter <b>IRS</b> as the name of the group.</li>
    <li>Repeat with the other complexes and groups, naming them <b>PI3K</b> and <b>AKT</b> respectively.</li>
    </ul>
    <p style="text-align: left">The gene set now includes three group nodes.</p>
    <img class="plain" src="gene-set-groups.png" style="width:85%">
</section>
    
<section>
    <h1>Add Interactions to the Gene Set</h1>
    <p style="text-align: left">Once the nodes are grouped, we can add and style the interactiosn between them.</p>
    <ul>
    <li>Select the IRS and PI3K group nodes by clicking on the group node outline. The outline should be highlighted in yellow.</li>
    <li>Right-click to access the context menu. Select <highlight>Add &rarr; Edges Connecting Selected Nodes</highlight>.</li>
    
    </ul>
    <img class="plain" src="gene-set-interactions1.png" style="width:85%">
</section>
<section>
    <h1>Style Interactions to the Gene Set</h1>
    <ul>
    <li>Select one of the edges. In the <highlight>Style</highlight> interface, select the <highlight>Edge</highlight> tab.</li>
    <li>For <highlight>Target Arrow Shape</highlight>, click on the <highlight>Bypass</highlight> column and select an arrow. Click <highlight>Apply</highlight>. Note that it is sometimes not clear which node is the source vs target, so for some interactions you may need to edit the <highlight>Source Arrow Shape</highlight>.</li>
    <li>Repeat this process to add all relevant interactions from the pathway figure to the gene set. </li>
    </ul>
    <p style="text-align: left">You may need to move the nodes around for optimal layout.</p>
    <img class="plain" src="gene-set-interactions2.png" style="width:85%">
    
</section>
    
</section>

<section>  
<section>
    <h1>Retreiving Interactions from STRING</h1>
        <p style="text-align: left">So far, we have worked with the gene set as nodes, without edges. Next, we will add <b>known interactions</b> to our nodes from the <a href="https://string-db.org/">STRING</a> database by quering the database for all nodes. This will enable further analysis options downstream.</p>
        <ul>
            <li>In the <highlight>Node Table</highlight>, select the top entry in the <highlight>name</highlight> column, hold down the shift key and select the last entry. Copy the contents to the clipboard.</li>
            <li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING protein query</b> in the drop-down and paste in the contents of the clipboard.</li>
            <li>Click the <highlight>More Options...</highlight> button <img class="plain" src="string-options.png" style="height:30px"> and change the <highlight>Confidence score cutoff</highlight> to <b>0.95</b>.</li>
        </ul>   
</section>
    
<section>
    <h1>Retreiving Interactions from STRING</h1>
    <p style="text-align: left">The resulting network contains the nodes from our pathway gene set, and interactions between them with an evidence score of 0.95 or greater. Adding a force-directed layout, the network looks something like this:</p>
    <img class="plain" src="pfocr-string-layout.png" >
    <!--<img class="plain" src="pfocr-string.png" style="width:55%"> -->
</section>
    
<!--
<section>
<h1>STRING Network</h1>
    <p>The network consists of one large connected component, several smaller networks, and some unconnected nodes. We will use only the largest connected component for the rest of the tutorial.</p>
    <ul>
    <li>Using Shift-drag, select all nodes and edges for the largest connected component.</li>
    <li>Select <highlight>File &rarr; New Network &rarr; From Selected Nodes, Selected Edges</highlight>.</li>
    </ul>
<img class="plain" src="string-de-up2-sub.png" style="height:600px"> 
</section>
-->
   
<section>
    <h1>Data Integration</h1>
    <p style="text-align: left">Next, we can import the same experimental data from a <a href="https://cancergenome.nih.gov/" target="_blank">TCGA</a> dataset. The STRING network is annotated with protein identifiers, mainly <b>Uniprot</b> and <b>Ensembl protein</b>, while the data has <b>NCBI Gene</b> identifiers. So before importing the data we are going to use the <b>ID Mapper</b> functionality in Cytoscape to map the network to NCBI Gene.</p>
    <ul>
        <li>In the <highlight>Node Table</highlight>, right-click on the column header of the <highlight>display name</highlight> column and click <highlight>Map column...</highlight>.</li>
        <li>In the <highlight>ID Mapping</highlight> interface, select <b>Human</b> as <highlight>Species</highlight>, <b>HGNC</b> as <highlight>Map from</highlight> and <b>Entrez</b> as <highlight>To</highlight>. Click <b>OK</b> to continue.</li>
        <li><b>IDMapper</b> displays a report of how many identifiers were mapped. Make note of this information as it impacts all downstream analysis; If the mapping was unsuccessful, downstream analysis will be as well.</li>
    </ul>
</section>
                
<section>
    <h1>Data Integration</h1>
    <ul>
        <li>Load the <b>TCGA-Ovarian-MesenvsImmuno_data.csv</b> file (downloaded earlier) under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>. Alternatively, drag and drop the data file directly onto the <highlight>Node Table</highlight>.</li>
	   <li>Select the new <b>Entrez Gene</b> column as the <b>Key column for Network</b> to match the <b>Gene</b>column of the data.</li>
        <li>To complete the import, click <highlight>OK</highlight>. Two new columns of data will be added to the <highlight>Node Table</highlight>.</li>
    </ul>
</section>
                
<section>
    <h1>Visualization</h1>
    <p style="text-align: left">Next, we will create a visualization of the imported data on the network. For more detailed information about how to create visualizations, see the <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/mapping-data/#/" target="_blank">Visualizing Data</a> tutorial.</p>   
    <ul>
    <li>In the <highlight>Style</highlight> tab of the <highlight>Control Panel</highlight>, delete the mappings for <highlight>Image/Chart 1</highlight> and <highlight>Image/Chart 2</highlight> by clicking the <highlight>Delete</highlight> button <img class="plain" src="Delete.png" style="height:30px"> for each.</li>
    <li>Set the default node fill color to light gray.</li>
    <li>Set the default <highlight>Border Width</highlight> to 2, and make the default <highlight>Border Paint</highlight> dark gray.</li>   
    <li>For node <highlight>Fill Color</highlight>, create a continuous mapping for <highlight>logFC</highlight>, with the default ColorBrewer <b>red-blue</b> gradient.</li>
	</ul>
</section> 
                
<section>
     <h1>Visualization</h1>
        <p style="text-align: left">The network will now look something like this:</p>
	    <img class="plain" src="pfocr-string-style.png" style="height:650px">  
</section>
                
<section>
    <h1>STRING Enrichment</h1>
    <p style="text-align: left">The STRING app has built-in enrichment analysis functionality, which includes enrichment for GO Process, GO Component, GO Function, InterPro, KEGG Pathways, and PFAM.</p>
    <ul>
    <li>In the STRING tab of the <highlight>Results Panel</highlight>, click the <highlight>Functional Enrichment</highlight> button. Keep the default settings.</li>
    <li>When the enrichment analysis is complete, a new tab titled <highlight>STRING Enrichment</highlight> will open in the <highlight>Table Panel</highlight>.</li>
    </ul>
<img class="plain" src="string-enrichment.png" style="height:300px">    
</section>
                
<section>
    <h1>STRING Enrichment</h1>
    <p style="text-align: left">The STRING app includes several options for filtering and displaying the enrichment results. The features are all available at the top of the <highlight>STRING Enrichment</highlight> tab. We are going to filter the table to only show GO Process.</p>    
    <ul>
        <li>At the top left of the STRING enrichment tab, click the filter icon <img class="plain" src="FilterIcon.png" style="height:30px">. Select <highlight>GO Process</highlight> and check the <highlight>Remove redundant terms</highlight> check-box. Then click <b>OK</b>.</li>
        <li>Next, we will add a split donut chart to the nodes representing the top terms by clicking on <img class="plain" src="ChartIcon.png" style="height:30px">.</li>
    </ul>
</section>
    
<section>
    <h1>STRING Enrichment</h1>
    <p style="text-align: left">Not surprisingly, the top two terms are related to the signal transduction, specifically phosphatidylinositol phosphorylation, the enzymatic function of PI3Ks.</p>
    <img class="plain" src="string-enrichment-chart.png" style="height:450px">    
    <img class="plain" src="string-enrichment2.png" style="width:1100px">    

    
</section>
                
<section>
    <h1>Add Pathway Figure as Annotation</h1>
    <p style="text-align: left">Again, we can add the original published figure for this gene set as an annotation.</p>
    <ul>
        <li>In the <highlight>Annotations</highlight> tab of the <highlight>Control Panel</highlight>, select the <highlight>Image</highlight> icon at the top and then click anywhere in the network view to place the annotation.</li>
        <li>Select the pathway figure you downloaded previously.</li>
        <li>Once the annotation is placed, enable annotation selection by clicking the <highlight>Toggle Annotation Selection</highlight> button <img class="plain" src="toggle-annotation.png" style="height:30px"> in the <highlight>Network View Tools</highlight> under the network.</li>
    </ul>
    <img class="plain" src="network-final.png" style="height:550px"> 
</section>

<!--<section>
    <p>We can see that the two subsets in the network correspond to the middle and right arm of the pathway, and the unconnected nodes at the bottom correspond roughly to downstream proteins.</p>
    
    <img class="plain" src="network-final.png" style="height:600px"> 
    
</section>-->
</section> 
                
<section>
<section>
    <h1>Retrieving interactions from INDRA</h1>
    <p></p>
    
    <p style="text-align: left">An alternative source for interactions for the PFOCR gene set is the <a href="http://www.indra.bio/">INDRA</a> database. INDRA contains interactions from multiple sources, with directional information. We will access the network from NDEx.</p>
    <ul>
        <li>In Cytoscape, select the original PFOCR node set imported from NDEx. In the <highlight>Node Table</highlight>, select the top entry in the <highlight>name</highlight> column, hold down the shift key and select the last entry. Copy the contents to the clipboard.</li>
        <li>Navigate to the <a href="https://ndexbio.org/viewer/networks/04020c47-4cfd-11e8-a4bf-0ac135e8bacf">INDRA database entry</a> at NDEx. The network is very large, 79,000+ nodes, and will not be displayed at NDEx.</li>
        <li>In the lower left under the network, in the <highlight>Enter query terms</highlight> field, paste the PFOCR nodes.</li>
        <li>In the drop-down, change the type of search to <b>Direct</b>.</li>
        <li>Click the search icon to search. Because the network is large, the search will take some time.</li> 
    </ul>
</section>
    
<section>
    <h1>Retrieving interactions from INDRA</h1>
    <ul>
        <li>Once the subnetwork loads, click the <highlight>Download query result as CX</highlight> button <img class="plain" src="ndex-download.png" style="height:30px">.</li>
        <li>In Cytoscape, go to <highlight>File &rarr; Import &rarr; Network from File...</highlight> and select the downloaded CX file.</li>
        <li>The network will look similar to this:</li>
    </ul>
    <img class="plain" src="pfocr_indra.png" style="height:600px">
</section>
    
<section>
    <h1>Retrieving interactions from INDRA</h1>
    <p>Let's clean up the network a bit:</p>
    <ul>
        <li>To remove self-loops and duplicated edges, go to <highlight>Edit &rarr; Remove Self-Loops....</highlight>, selecting the newly created network.</li>
        <li>Next, go to <highlight>Edit &rarr; Remove Duplicate Edges....</highlight> and again select the right network. Leave the two options unchecked.</li>
        <li>Finally, delete the two unconnected nodes.</li>
    </ul>
    <img class="plain" src="pfocr_indra_clean.png" style="height:500px">
</section>
    
<section>
    <h1>Data Integration</h1>
    <p style="text-align: left">Next, we can import the same experimental data from a <a href="https://cancergenome.nih.gov/" target="_blank">TCGA</a> dataset. The INDRA network, and therefor our new subnetwork, is annotated with Gene symbols, while the data has <b>NCBI Gene</b> identifiers. So before importing the data we are going to use the <b>ID Mapper</b> functionality in Cytoscape to map the network to NCBI Gene.</p>
    <ul>
        <li>In the <highlight>Node Table</highlight>, right-click on the column header of the <highlight>name</highlight> column and click <highlight>Map column...</highlight>.</li>
        <li>In the <highlight>ID Mapping</highlight> interface, select <b>Human</b> as <highlight>Species</highlight>, <b>HGNC</b> as <highlight>Map from</highlight> and <b>Entrez</b> as <highlight>To</highlight>. Click <b>OK</b> to continue.</li>
        <li><b>IDMapper</b> displays a report of how many identifiers were mapped. Make note of this information as it impacts all downstream analysis; If the mapping was unsuccessful, downstream analysis will be as well.</li>
    </ul>
</section>
    
<section>
    <h1>Data Integration</h1>
    <ul>
        <li>Load the <b>TCGA-Ovarian-MesenvsImmuno_data.csv</b> file (downloaded earlier) under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>. Alternatively, drag and drop the data file directly onto the <highlight>Node Table</highlight>.</li>
	   <li>Select the new <b>Entrez Gene</b> column as the <b>Key column for Network</b> to match the <b>Gene</b>column of the data.</li>
        <li>To complete the import, click <highlight>OK</highlight>. Two new columns of data will be added to the <highlight>Node Table</highlight>.</li>
    </ul>
</section>
    
<section>
    <h1>Visualization</h1>
    <p style="text-align: left">Next, we will create a visualization of the imported data on the network. For more detailed information about how to create visualizations, see the <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/mapping-data/#/" target="_blank">Visualizing Data</a> tutorial.</p>   
    <ul>
    <li>Set the default <highlight>Border Width</highlight> to 2, and make the default <highlight>Border Paint</highlight> dark gray.</li>   
    <li>For node <highlight>Fill Color</highlight>, create a continuous mapping for <highlight>logFC</highlight>, with the default ColorBrewer <b>red-blue</b> gradient.</li>
	</ul>
    
    <img class="plain" src="pfocr_indra_viz1.png" style="height:500px">
</section> 
    
<section>
    <h1>Visualization</h1>
    <p style="text-align: left">We can also add visualization to the edges, since the INDRA database provides information on the type of interaction. We will use a combination of arrow heads and interaction color.</p>   
    <ul>
        <li>Navigate to the <highlight>Edge</highlight> tab of the <highlight>Style</highlight> panel.</li>
        <li>For <highlight>Target Arrow Shape</highlight> create a <b>Discrete</b> mapping for <b>type</b>. This allows you to pick a different arrow head for each interaction type in the network. The selection is done by clicking on a particular entry, for example <b>Activation</b>, and selecting an arrow type in the popup.
        <ul>
            <li>For <b>Activation</b>, <b>Modification</b> and <b>IncreaseAmount</b>, select the <b>Delta</b> arrow head.</li>
            <li>For <b>Inhibition</b>, select the the <b>T</b>.</li>
            <li>For <b>DecreaseAmount</b>, select <b>Cross Delta</b>.</li>
        </ul>
        </li>
        
	</ul>
</section> 
    
<section>
     <h1>Visualization</h1>
    <ul>
    <li>For the final interaction type, <b>Complex</b>, go to the <highlight>Stroke Color</highlight> entry in the edge tab, and again create a <b>Discrete</b> mapping for <b>type</b>. Choose a green color for <b>Complex</b>, and leave the others blank.</li>
    </ul>
    
        <img class="plain" src="pfocr_indra_viz2.png" style="height:500px">

</section>
    
<section>
    <h1>Manual layout</h1>
    <p>We can manually arrange the nodes to match the layout of the original figure. To beging, we can import the figure as an annotation.</p>
    <ul>
        <li>In the <highlight>Annotations</highlight> tab of the <highlight>Control Panel</highlight>, select the <highlight>Image</highlight> icon at the top and then click anywhere in the network view to place the annotation.</li>
        <li>Select the pathway figure you downloaded previously.</li>
        <li>Once the annotation is placed, enable annotation selection by clicking the <highlight>Toggle Annotation Selection</highlight> button <img class="plain" src="toggle-annotation.png" style="height:30px"> in the <highlight>Network View Tools</highlight> under the network.</li>
        <li>To mimic the layout, simple click and drag the nodes around to make the network look somethine like this:</li>
    </ul>
    
    <img class="plain" src="pfocr_indra_viz3.png">
</section>

<section>
    <h1>Manual layout</h1>
    <p>Once we group the nodes, we can see that there is an extra <b>Akt1</b> node. Since the gene name is not upper-cased (the convention for human), we can assume that this node is for a different species, in this case Drosophila.</p>
    <ul>
    <li>Delete the <b>Akt1</b> node by selecting it and clicking the <b>delete</b> key.</li>
    
    </ul>
    <img class="plain" src="pfocr_indra_viz4.png">
</section>
    
</section>
                
<section>
	<h1>Saving Results</h1>
    <p>You now have a cytoscape network representing the original pathway figure, that can be used for further analysis, to  visualiza additional data and to create figures for presentations or publications. Cytoscape provides a number of ways to save results and visualizations:</p>
	<ul>
        <li>Save your results as a session file: <highlight>File &rarr; Save</highlight>, <highlight>File &rarr; Save As...</highlight></li>
        <li>Export as an image: <highlight>File &rarr; Export &rarr; Network to Image...</highlight></li>
        <li>Export to the web: <highlight>File &rarr; Export &rarr; Network to Web Page...</highlight> (<a href="../saving-results/web_session/index.html" target="_blank">Example</a>)</li>
        <li>Export to a public repository: <highlight>File &rarr; Export &rarr; Network to NDEx</highlight>, or <highlight>File &rarr; Export &rarr; Collection to NDEx</highlight></li>
		<li>As a graph format file: <highlight>File &rarr; Export &rarr; Network to File</highlight>. <br/>Formats:
						<ul>
										<li>CX JSON</li>
										<li>Cytoscape.js JSON</li>
										<li>GraphML</li>
										<li>PSI-MI</li>
										<li>XGMML</li>
										<li>SIF</li>
                        </ul></li>
	</ul>
</section>
                

            
            
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>  
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },
				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
