<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Pathway Figure Gene Set Analysis</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">

<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section>
	<h1>Pathway Figure Gene Set Analysis</h1>
    <p>Through a combination of optical character recognition, machine learning and manual curation, gene sets have been extracted from <a href="https://doi.org/10.1186/s13059-020-02181-2" target="_blank">published pathway figures</a>. 32,000 of these sets are available in <a href="https://ndexbio.org/#/networkset/85034b42-de8a-11ea-99da-0ac135e8bacf" target="_blank">NDEx</a>, and can be used directly in Cytoscape.<br />
    Starting with a gene set, this protocol describes how to visualize data on the set, create a network from it, perform functional enrichment and include the original figure for comparison.</p>
    
    <img class="plain" src="network-final.png" style="height:320px"> 
    
    <p>If you don't already have the <b>stringApp</b> installed, you can install it from the <a href="http://apps.cytoscape.org/apps/stringapp" target="_blank">Cytoscape App Store</a> or from Cytoscape via <highlight>Apps &rarr; App Manager ...</highlight></p>
    <div class="footer" style="position: absolute; left: -4em; bottom: -3em; ">
<a href="https://github.com/cytoscape/cytoscape-tutorials/wiki"><img class="plain" src="../../assets/images/cytoscape-tutorials.png" style="height:70px;opacity:0.5"></a>
</div>
</section>
                
<section>
    <h1>Import Pathway Figure Gene Set</h1>
    
    <ul>
        <li>Navigate to <a href="https://ndexbio.org/#/networkset/85034b42-de8a-11ea-99da-0ac135e8bacf" target="_blank">Published Pathway Figure Analysis Set</a> at NDEx.</li>
        <li>In the <highlight>Network Name</highlight> search field, search for <b>03f5f61a-df06-11ea-99da-0ac135e8bacf</b> to find the network by <highlight>UUID</highlight>. Click on the network. The page will display the genes in the set on the left and the published figure they were extracted from on the right.</li>
        <li>Save the figure by right-clicking on it and selecting <highlight>Save Image As..</highlight>.</li>
        <li>To open the gene set into Cytoscape, click the <highlight>Open in Cytoscape</highlight> button in the lower right.</li>
    </ul>
</section>
                
<!--<section>
    <h1>Import NDEx Gene Set</h1>
    <p>We will start by importing a gene set from NDEx.</p>
    <ul>
        <li>Click the <highlight>Import Network from NDEx</highlight> <img class="plain" src="NDEx-search.png" style="height:20px"> icon in the control panel.</li>
        <li>In the NDEx <highlight>Find Networks</highlight> interface, type in the network UUID <b>bcd5afe8-de97-11ea-99da-0ac135e8bacf</b> and click the search icon.</li>
        <li>When the results load, double-click the corresponding entry to open the network, in this case the network with name <b>PMC5884799__41598_2018_24032_Fig7_HTML.jpg</b>.</li>
        <li>Close the NDEx interface.</li>
    </ul>
    <img class="plain" src="NDEx-results.png" style="width:45%">
</section>-->

<section>
<section>
    <h1>Data Integration and Visualization</h1>
    <p>We are going to import experimental data from a <a href="https://cancergenome.nih.gov/" target="_blank">TCGA</a> dataset, comparing gene expression between two subtypes of ovarian cancer. The data is annotated with <b>NCBI Gene</b> identifiers, which is also what the nodes in the gene set are annotated with.</p>
    <ul>
        <li>Download a local copy of <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/data/TCGA-Ovarian-MesenvsImmuno_data.csv" target="_blank">TCGA-Ovarian-MesenvsImmuno_data.csv</a>.</li>
        <li>Load the <b>TCGA-Ovarian-MesenvsImmuno_data.csv</b> file under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight>. Alternatively, drag and drop the data file directly onto the <highlight>Node Table</highlight>.</li>
	   <li>Select the <b>ncbigene</b> column as the <b>Key column for Network</b> to match the correct column with the key column of the data.</li>
        <li>To complete the import, click <highlight>OK</highlight>. Two new columns of data will be added to the <highlight>Node Table</highlight>.</li>
    </ul>
</section>   
                
<section>
    <h1>Data Integration and Visualization</h1>
    <p>We can now add a simple visualization of the data on the set of nodes.</p>
    <ul>
        <li>In the <highlight>Style</highlight> interface, under <highlight>Node Fill Color</highlight>, create a continuous mapping for <highlight>logFC</highlight>, with the default ColorBrewer <b>Red-Blue</b> gradient.</li>
        <li>Set the default node fill color to light gray.</li>
        <li>The gene set is already arranged with a grid layout, but we can make it a bit more compact. Under <highlight>Layout &rarr; Settings...</highlight>, select the <highlight>Grid Layout</highlight>.</li>
        <li>Set the <highlight>Vertical spacing between nodes</highlight> to <b>50</b> and <highlight>Horizontal spacing between nodes</highlight> to <b>80</b>.</li>
    </ul>
    <img class="plain" src="pfocr-gene-set.png" style="width:50%">
</section>
                
<section>
    <h1>Add Pathway Figure as Annotation</h1>
    <p>Finally, we can add the original published figure for this gene set as an annotation, to add context to the nodes.</p>
    <ul>
        <li>In the <highlight>Annotations</highlight> tab of the <highlight>Control Panel</highlight>, select the <highlight>Image</highlight> icon at the top and then click anywhere in the network view to place the annotation. The initial placement can easily be changed later.</li>
        <li>Select the pathway figure you downloaded previously in the file browser.</li>
        <li>Once the annotation is placed, enable annotation selection by clicking the <highlight>Toggle Annotation Selection</highlight> button <img class="plain" src="toggle-annotation.png" style="height:30px"> in the <highlight>Network View Tools</highlight> under the network.</li>
    </ul>
    <img class="plain" src="gene-set-final.png" style="width:85%">
</section>
    
<section>
    <h1>Add Pathway Figure as Annotation</h1>
    <p><b>Bonus:</b> If we arrange the nodes in the gene set to approximate their placement in the original figure, we can begin to approximate a pathway layout, and start to interpret the data. Note that the label displayed on the node, <b>name</b>, is the official gene symbol, whereas the original published figures sometimes use various gene symbol aliases. Use the <b>word</b> column to match nodes to the figure.</p>
    
        <img class="plain" src="gene-set-final-pw.png" style="width:85%">

    
</section>
</section>

<section>  
<section>
    <h1>STRING Query</h1>
        <p>So far, we have worked with the gene set as nodes, without edges. Next, we will add <b>known interactions</b> to our nodes from the STRING database by quering the database for all nodes. This will enable further analysis options downstream.</p>
        <ul>
            <li>In the <highlight>Node Table</highlight>, select the top entry in the <highlight>name</highlight> column, hold down the shift key and select the last entry. Copy the contents to the clipboard.</li>
            <li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING protein query</b> in the drop-down and paste in the contents of the clipboard.</li>
            <li>Click the <highlight>More Options...</highlight> button <img class="plain" src="string-options.png" style="height:30px"> and change the <highlight>Confidence score cutoff</highlight> to <b>0.95</b>.</li>
        </ul>   
</section>
    
<section>
    <h1>STRING Query</h1>
    <p>The resulting network contains the nodes from our pathway gene set, and interactions between them with an evidence score of 0.95 or greater. Adding a force-directed layout we can begin to see two connected subsets:</p>
    <img class="plain" src="pfocr-string-layout.png" style="width:43%">
    <!--<img class="plain" src="pfocr-string.png" style="width:55%"> -->
</section>
    
<!--
<section>
<h1>STRING Network</h1>
    <p>The network consists of one large connected component, several smaller networks, and some unconnected nodes. We will use only the largest connected component for the rest of the tutorial.</p>
    <ul>
    <li>Using Shift-drag, select all nodes and edges for the largest connected component.</li>
    <li>Select <highlight>File &rarr; New Network &rarr; From Selected Nodes, Selected Edges</highlight>.</li>
    </ul>
<img class="plain" src="string-de-up2-sub.png" style="height:600px"> 
</section>
-->
   
<section>
    <h1>Data Integration</h1>
    <p>Next, we can import the same experimental data from a <a href="https://cancergenome.nih.gov/" target="_blank">TCGA</a> dataset. The STRING network is annotated with protein identifiers, mainly <b>Uniprot</b> and <b>Ensembl protein</b>, while the data has <b>NCBI Gene</b> identifiers. So before importing the data we are going to use the <b>ID Mapper</b> functionality in Cytoscape to map the network to NCBI Gene.</p>
    <ul>
        <li>In the <highlight>Node Table</highlight>, right-click on the column header of the <highlight>display name</highlight> column and click <highlight>Map column...</highlight>.</li>
        <li>In the <highlight>ID Mapping</highlight> interface, select <b>Human</b> as <highlight>Species</highlight>, <b>HGNC</b> as <highlight>Map from</highlight> and <b>Entrez</b> as <highlight>To</highlight>. Click <b>OK</b> to continue.</li>
        <li><b>IDMapper</b> displays a report of how many identifiers were mapped. Make note of this information as it impacts all downstream analysis; If the mapping was unsuccessful, downstream analysis will be as well.</li>
    </ul>
</section>
                
<section>
    <h1>Data Integration</h1>
    <ul>
        <li>Load the <b>TCGA-Ovarian-MesenvsImmuno_data.csv</b> file (downloaded earlier) under <highlight>File</highlight> menu, select <highlight>Import &rarr; Table from File....</highlight></li>
	   <li>Make sure you select the new <b>Entrez Gene</b> column as the <b>Key column for Network</b> to match the correct column with the key column of the data.</li>
        <li>To complete the import, click <highlight>OK</highlight>. Two new columns of data will be added to the <highlight>Node Table</highlight>.</li>
    </ul>
</section>
                
<section>
    <h1>Visualization</h1>
    <p>Next, we will create a visualization of the imported data on the network. For more detailed information about how to create visualizations, see the <a href="https://cytoscape.github.io/cytoscape-tutorials/protocols/mapping-data/#/" target="_blank">Visualizing Data</a> tutorial.</p>   
    <ul>
    <li>In the <highlight>Style</highlight> tab of the <highlight>Control Panel</highlight>, switch the style from <b>STRING style v1.5</b> to <b>default</b> in the drop-down at the top.</li>
    <li>Change the default node shape to <b>ellipse</b> and check <b>Lock node width and height</b>.</li>
    <li>Set the default node size to 50.</li>
    <li>Set the default node fill color to light gray.</li>
    <li>Set the default <highlight>Border Width</highlight> to 2, and make the default <highlight>Border Paint</highlight> dark gray.</li>   
    <li>For node <highlight>Fill Color</highlight>, create a continuous mapping for <highlight>logFC</highlight>, with the default ColorBrewer <b>yellow-orange-red shades</b> gradient.</li>
    <li>For <highlight>Node Label</highlight>, set a passthrough mapping for <highlight>display name</highlight>.</li>
	</ul>
</section> 
                
<section>
     <h1>Visualization</h1>
        <p>The network will now look something like this:</p>
	    <img class="plain" src="pfocr-string-style.png" style="height:650px">  
</section>
                
<section>
    <h1>STRING Enrichment</h1>
    <p>The STRING app has built-in enrichment analysis functionality, which includes enrichment for GO Process, GO Component, GO Function, InterPro, KEGG Pathways, and PFAM.</p>
    <ul>
    <li>In the STRING tab of the <highlight>Results Panel</highlight>, click the <highlight>Functional Enrichment</highlight> button. Keep the default settings.</li>
    <li>When the enrichment analysis is complete, a new tab titled <highlight>STRING Enrichment</highlight> will open in the <highlight>Table Panel</highlight>.</li>
    </ul>
<img class="plain" src="string-enrichment.png" style="height:300px">    
</section>
                
<section>
    <h1>STRING Enrichment</h1>
    <p>The STRING app includes several options for filtering and displaying the enrichment results. The features are all available at the top of the <highlight>STRING Enrichment</highlight> tab. We are going to filter the table to only show GO Process.</p>    
    <ul>
        <li>At the top left of the STRING enrichment tab, click the filter icon <img class="plain" src="FilterIcon.png" style="height:30px">. Select <highlight>GO Process</highlight> and check the <highlight>Remove redundant terms</highlight> check-box. Then click OK.</li>
        <li>Next, we will add a split donut chart to the nodes representing the top terms by clicking on <img class="plain" src="ChartIcon.png" style="height:30px">.</li>
    </ul>
<img class="plain" src="string-enrichment-chart.png" style="height:500px">    
</section>
                
<section>
    <h1>Add Pathway Figure as Annotation</h1>
    <p>Again, we can add the original published figure for this gene set as an annotation.</p>
    <ul>
        <li>In the <highlight>Annotations</highlight> tab of the <highlight>Control Panel</highlight>, select the <highlight>Image</highlight> icon at the top and then click anywhere in the network view to place the annotation. The initial placement can easily be changed later.</li>
        <li>Select the pathway figure you downloaded previously in the file browser.</li>
        <li>Once the annotation is placed, enable annotation selection by clicking the <highlight>Toggle Annotation Selection</highlight> button <img class="plain" src="toggle-annotation.png" style="height:30px"> in the <highlight>Network View Tools</highlight> under the network.</li>
    </ul>
</section>

<section>
    <p>We can see that the two subsets in the network correspond to the middle and right arm of the pathway, and the unconnected nodes at the bottom correspond roughly to downstream proteins.</p>
    
    <img class="plain" src="network-final.png" style="height:600px"> 
    
</section>
</section>          
                

            
            
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>  
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },
				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
